gossip/src/duplicate_shred_handler.rs

```diff
@@ -90,17 +97,42 @@ impl DuplicateShredHandler {
     pub fn new(
         blockstore: Arc<Blockstore>,
         leader_schedule_cache: Arc<LeaderScheduleCache>,
+        bank_forks: Arc<RwLock<BankForks>>,
     ) -> Self {
         Self {
             chunk_map: HashMap::new(),
             validator_pending_proof_map: HashMap::new(),
             last_root: 0,
+            last_root_cleaned: 0,
+            cached_on_slot: 0,
+            cached_staked_nodes: None,
+            cached_slots_in_epoch: 100,
```

Comment:
As said above, I don't want to do read operations in initialization. 100 is a random number I put in there just in case we can't read root bank for a while, changed to 0 now.